### PR TITLE
add further input check of batches and number of entries

### PR DESCRIPTION
**Pull request by Leon-Bichmann**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
 - [x] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
 - [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/diaproteomics/tree/master/.github/CONTRIBUTING.md)
 - [x] If necessary, also make a PR on the nf-core/diaproteomics _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by Leon-Bichmann** [comment]
fixes #73

File: main.nf [hunk 1]
```diff
@@ -182,6 +182,11 @@ if( params.generate_spectral_library) {
         exit 1
     }
 
+    if ((check_n_sample != check_dia_n)) {
+        print('The number of batches in the sample input do not match the number of batches of the spectral library input \n')
```
Comment (lkuchenb) [inline main.nf]:
```suggestion
        print('The number of batches in the sample input does not match the number of batches of the spectral library input \n')
```

File: main.nf [hunk 2]
```diff
@@ -231,7 +249,20 @@ if( params.generate_spectral_library) {
...
+    check_lib_n2 = input_lib_n2.toList().size().val
+    check_lib_n = input_lib_n.map{it[1]}.unique().toList().size().val
+    if ((check_lib_n > 1) & (check_lib_n2 != check_lib_n) & (!params.merge_libraries)) {
+        print('You specified multiple spectral libraries for one batch \n')
+        print('Set --merge_libraries and possibly --align_libraries to align them in the same RT space \n')
+        exit 1
+    }
+
+    if ((check_lib_n != check_dia_n)) {
+        print('The number of batches in the sample input do not match the number of batches of the spectral library input \n')
```
Comment (lkuchenb) [inline main.nf]:
```suggestion
        print('The number of batches in the sample input does not match the number of batches of the spectral library input \n')
```

File: main.nf [hunk 3]
```diff
@@ -254,7 +285,20 @@ if( !params.generate_pseudo_irts){
...
+    check_irts_n = input_irts_check.toList().size().val
+    check_irts_n2 = input_irts_check_2.map{it[0]}.unique().toList().size().val
+    if ((check_irts_n > 1) & (check_irts_n != check_irts_n2)) {
+        print('You specified multiple DDA files to generate spectral libraries, but library merging is not set \n')
+        print('Set --merge_libraries and possibly --align_libraries to align them in the same RT space \n')
+        exit 1
+    }
+
+    if ((check_irts_n2 != check_dia_n)) {
+        print('The number of batches in the sample input do not match the number of batches of the irt input \n')
```
Comment (lkuchenb) [inline main.nf]:
```suggestion
        print('The number of batches in the sample input does not match the number of batches of the irt input \n')
```

**Review by lkuchenb — CHANGES_REQUESTED** [review]
👍 the more user feedback the better

**Pull request merged**